drgn_tools/iscsi.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2024, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper for iscsi
+"""
+import argparse
+from typing import Iterator
+
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common import escape_ascii_string
+from drgn.helpers.common.type import enum_type_to_class
+from drgn.helpers.linux.list import list_for_each_entry
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.module import ensure_debuginfo
+from drgn_tools.scsi import for_each_scsi_host
+from drgn_tools.scsi import for_each_scsi_host_device
+from drgn_tools.scsi import host_module_name
+from drgn_tools.scsi import scsi_device_name
+from drgn_tools.table import print_dictionary
+from drgn_tools.util import enum_name_get
+
+ISCSI_SESSION_STATES = ["LOGGED_IN", "FAILED", "FREE"]
+
+
+def for_each_iscsi_host(prog: Program) -> Iterator[Object]:
+    """
+    Iterate through all scsi host devices and returns an
+    iterator of hosts backed by iscsi drivers.
+    :returns: an iterator of ``struct Scsi_Host *``
+    """
+
+    # additional iscsi drivers
+    ISCSI_MODS_ADD = [
+        "qla4xxx",
+        "libcxgbi",
+        "bnx2i",
+        "cxgb3",
+        "cxgb3i",
+        "cxgb4",
+        "cxgb4i",
+    ]
```

Comment:
`libcxgbi` is a dependency module of `cxgb3i` and `cxgb4i`, we don't need it? And also `cxgb3` and `cxgb4` not a kernel module name?
`qedi` is also iscsi driver, please add.
Please add `iscsi_tcp` and `be2iscsi` in the list, so you don't need check whether iscsi is in the name of driver module below.